lib/config.js

```diff
@@ -867,6 +867,18 @@ util.parseFile = function(fullFilename) {
     if (extension === 'js') {
       // Use the built-in parser for .js files
       configObject = require(fullFilename);
+    } 
+    else if (extension === 'ts') {
+      require('ts-node').register({
```

Comment:
Is it really necessary to register ts-node here? Having config in TS most probably means that the entire project is written in TS, so ts-node most probably has already been registered anyway. 

I guess this line is here to make the tests pass. If so, isn't it better to register ts-node at test runner's bootstrap?

I actually have a use case where this ts-node registration here is a real stopper. It happens in a typescript project which uses ts-node. A couple of modules in this project have circular dependencies. The final result is that ts-node report extremely weird (and completely non-existent!) typescript compilation errors. Commenting out this statement resolves my problem.

I'll do my best to provide a minimal example demonstrating the problem.

Reply:
If this is still a problem, open a new issue.